packages/mermaid/src/Diagram.ts

```diff
@@ -5,6 +5,7 @@ import { detectType, getDiagramLoader } from './diagram-api/detectType.js';
 import { UnknownDiagramError } from './errors.js';
 import type { DetailedError } from './utils.js';
 import type { DiagramDefinition, DiagramMetadata } from './diagram-api/types.js';
+import { encodeEntities } from './mermaidAPI.js';
```

Comment:
> As per the lint rule failing for circular deps, I think we can move the util encodeEntities to packages/mermaid/src/utils.ts since now its a reusable common util. let me know If I should move the util.

Yes, this should be moved. 
Maybe you can move both encode and decode functions into it's own file.